src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSAppCommand.cs

```diff
@@ -159,24 +160,49 @@ private async Task<ExitCode> RunApp(ILogger logger, TestTargetOs target, Logs lo
                 return ExitCode.FAILED_TO_GET_BUNDLE_INFO;
             }
 
+            ExitCode exitCode = ExitCode.SUCCESS;
+
             if (!target.Platform.IsSimulator())
             {
-                var result = await InstallApp(
-                    appBundleInfo,
-                    deviceName,
-                    logger,
-                    target,
-                    logs,
-                    mainLog,
-                    cancellationToken);
+                Exception? installException = null;
 
-                if (result != ExitCode.SUCCESS)
+                try
                 {
-                    return result;
+                    (deviceName, exitCode) = await InstallApp(appBundleInfo, deviceName, logger, target, mainLog, cancellationToken);
                 }
-            }
+                catch (Exception e)
+                {
+                    installException = e;
+                }
+
+                if (exitCode != ExitCode.SUCCESS || installException != null)
+                {
+                    var message = new StringBuilder().AppendLine("Application installation failed:");
```

Comment:
Wouldn't it be worth to extract error message logic? It's there twice with almost the same code.

Reply:
I tried but the code is different enough (we use different knowledge base method, we get the exception from a different place, in one case we don't care about the exception message...) that it would be just too much overhead to add some common method for this.